### PR TITLE
Batch june rest2

### PR DESCRIPTION
**Pull request by bgklein**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/batch/_completers.py [hunk 1]
```diff
@@ -18,14 +18,14 @@ def load_node_agent_skus(cmd, prefix, namespace):  # pylint: disable=unused-argu
     client_creds['account_endpoint'] = cmd.cli_ctx.config.get('batch', 'endpoint', None)
     try:
         client = account_client_factory(cmd.cli_ctx, client_creds)
-        skus = client.list_node_agent_skus()
+        skus = client.list_supported_images()
         for sku in skus:
-            for image in sku['verifiedImageReferences']:
+            if sku.verification_type == "verified":
```
Comment (xingwu1) [inline src/azure-cli/azure/cli/command_modules/batch/_completers.py]: use "verification eq 'verified'" as filter

Comment (xingwu1) [reply src/azure-cli/azure/cli/command_modules/batch/_completers.py]: Also change the function name.

File: src/azure-cli/azure/cli/command_modules/batch/_completers.py [hunk 2]
```diff
@@ -18,14 +18,14 @@ def load_node_agent_skus(cmd, prefix, namespace):  # pylint: disable=unused-argu
     client_creds['account_endpoint'] = cmd.cli_ctx.config.get('batch', 'endpoint', None)
     try:
         client = account_client_factory(cmd.cli_ctx, client_creds)
-        skus = client.list_node_agent_skus()
+        skus = client.list_supported_images()
```
Comment (xingwu1) [inline src/azure-cli/azure/cli/command_modules/batch/_completers.py]: change this function name.

File: src/azure-cli/azure/cli/command_modules/batch/_completers.py (hunk 1 above)
Comment (xingwu1) [inline src/azure-cli/azure/cli/command_modules/batch/_completers.py]: remove this check. We should allow not validated image to create by cli.

File: src/azure-cli/azure/cli/command_modules/batch/commands.py [hunk 3]
```diff
@@ -82,8 +82,8 @@ def get_data_factory(name):
         g.batch_command('list', 'list')
         g.batch_command('show', 'get')
 
-    with self.command_group('batch pool node-agent-skus', get_data_type('account')) as g:
-        g.batch_command('list', 'list_node_agent_skus')
+    with self.command_group('batch pool supported-images', get_data_type('account')) as g:
+        g.batch_command('list', 'list_supported_images')
```
Comment (xingwu1) [inline src/azure-cli/azure/cli/command_modules/batch/commands.py]: Could you add a table transformer for this api?

**Comment by bgklein** [comment]
@marstr Is there anywhere else we need to edit to allow the image based tests to pass? Seems to be a problem since there is a major version dependency conflict.

**Comment by marstr** [comment]
> @bgklein says:
> @marstr Is there anywhere else we need to edit to allow the image based tests to pass? Seems to be a problem since there is a major version dependency conflict.

First thing to update is the requirements.txt files in [src/azure-cli/](https://github.com/Azure/azure-cli/tree/2aadf032ae306a41c692c18c871bdc9369f0987f/src/azure-cli). Once that's happened, I'll help investigate remaining failures to make sure their not systemic on our end. :)

**Comment by bgklein** [comment]
@xingwu1 please approve so CLI team can merge

**Comment by bgklein** [comment]
@marstr / @tjprescott if either of you want to take a look at this (specifically the README/setup/requirements files since those are recently reformatted) please review by EOD Friday and feel free to merge if you do. Otherwise I will merge after some traveling Friday.

**Pull request merged**